paradox-nakadi-consumer-sqs-error-handler/src/main/java/de.zalando.paradox.nakadi.consumer.sqserrorhandler/SQSErrorHandler.java

```diff
@@ -0,0 +1,79 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SQSErrorHandler.class);
+
+    private final String queueName;
+
+    private final AmazonSQS amazonSQS;
+
+    private final ObjectMapper objectMapper;
+
+    public SQSErrorHandler(final SQSConfiguration sqsConfiguration, final AmazonSQS amazonSQS,
+            final ObjectMapper objectMapper) {
+        this.amazonSQS = amazonSQS;
+        this.queueName = sqsConfiguration.getQueueName();
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public void onError(final String consumerName, final Throwable t, final EventTypePartition eventTypePartition,
+            final String offset, final String rawEvent) {
+
+        try {
+
+            checkArgument(isNotEmpty(consumerName), "consumerName must not be empty.");
+            checkArgument(isNotEmpty(rawEvent), "rawEvent must not be empty.");
+            checkArgument(isNotEmpty(offset), "offset must not be empty.");
+            checkNotNull(eventTypePartition, "eventPartition must not be null.");
+            checkNotNull(eventTypePartition.getEventType(), "eventTypePartition.getEventType() must not be null.");
+            checkArgument(isNotEmpty(eventTypePartition.getEventType().getName()), "eventName must not be null.");
+            checkArgument(isNotEmpty(eventTypePartition.getPartition()),
+                "eventTypePartition.getPartition() must not be null.");
+            checkNotNull(t, "exception must not be null.");
+
+            final FailedEvent failedEvent = new FailedEvent.Builder().consumerName(consumerName)
+                                                                     .eventType(eventTypePartition.getEventType())
+                                                                     .failedTimeInMilliSeconds(System
+                                                                             .currentTimeMillis()).offset(offset)
+                                                                     .partition(eventTypePartition.getPartition())
+                                                                     .rawEvent(rawEvent).throwable(t).build();
+
+            final String serializedEvent = objectMapper.writeValueAsString(failedEvent);
+            final GetQueueUrlResult queueUrl = amazonSQS.getQueueUrl(queueName);
+            final SendMessageResult sendMessageResult = amazonSQS.sendMessage(new SendMessageRequest(
+                        queueUrl.getQueueUrl(), serializedEvent));
+
+            if (sendMessageResult.getSdkHttpMetadata().getHttpStatusCode() != 200) {
+                LOGGER.error(
+                    "The result of sending event to SQS is not successful // Event body = [{}] , HttpStatusCode = [{}]",
+                    serializedEvent, sendMessageResult.getSdkHttpMetadata().getHttpStatusCode());
+                throw new IllegalStateException("The result of sending event to SQS is not successful.");
```

Comment:
Exception should carry all the information needed to debug. Logging error here will not be needed then.